src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -342,3 +423,384 @@ def test_WINNF_FT_S_PCR_1(self, config_filename):
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    conditionals = [conditionals_b]
+    del device_b['installationParam']
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['measCapability']
+
+    # Create the actual config.
+    devices = [device_a, device_b, device_c]
+    config = {
+        'registrationRequests': devices,
+        'conditionalRegistrationData': conditionals,
+        'palRecords': pal_records
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_4_default_config)
+  def test_WINNF_FT_S_PCR_4(self, config_filename):
+    """Unsuccessful PPA Creation with one or more CBSDs Outside Service Area.
+
+    Checks SAS UUT rejects creation of a PPA boundary if at least one of the CBSDs
+    included in the CBSD cluster list is located outside PAL holder service area.
+    """
+    # Load the Config file.
+    config = loadConfig(config_filename)
+
+    # Inject the PAL records.
+    for pal_record in config['palRecords']:
+      self._sas_admin.InjectPalDatabaseRecord(pal_record)
+
+    # Register devices and check the response.
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
+
+    # Trigger SAS UUT to create a PPA boundary.
+    pal_ids = [record['palId'] for record in config['palRecords']]
+
+    # Create PPA creation request to SAS UUT.
+    ppa_creation_request = {
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids
+    }
+
+    # Trigger PPA Creation to SAS UUT and expect the failure response.
+    self.assertPpaCreationFailure(ppa_creation_request)
+
+  def generate_PCR_5_default_config(self, filename):
+    """Generate the WinnForum configuration for PCR 5."""
+    # Load PAL records.
+    pal_record_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_record_1.json')))
+    pal_record_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_record_2.json')))
+
+    # Set the values of fipsCode in pal_records_a and b to make them adjacent.
+    # 20063955100 and 20063955200 respectively.
+    pal_record_a['fipsCode'] = 20063955100
+    pal_record_b['fipsCode'] = 20063955200
+
+    # Set the PAL frequency.
+    pal_low_frequency = 3570000000
+    pal_high_frequency = 3580000000
+
+    # Load device info.
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # Set the same user ID for all devices
+    device_b['userId'] = device_a['userId']
+
+    # Set the values of fipsCode in pal_records_a and b to make them adjacent.
+    # 20063955100 and 20063955200 respectively.
+    pal_records_a = makePalRecordsConsistent([pal_record_a], pal_low_frequency,
```

Comment:
I think you slightly misinterpreted: the PAL records should have the same user ID, but one of the CBSDs will have a different user ID.

So if you remove line 559 and set these both to device A's CBSD ID, that would be correct.

Reply:
Thanks.   line no 559 has been removed and updated the same device A's CBSD ID to pal   records.